prow/moonraker/moonraker.go

```diff
@@ -0,0 +1,83 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package moonraker
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/sirupsen/logrus"
+
+	prowapi "k8s.io/test-infra/prow/apis/prowjobs/v1"
+	"k8s.io/test-infra/prow/config"
+)
+
+const (
+	PathGetInrepoconfig = "inrepoconfig"
+)
+
+type Moonraker struct {
+	ConfigAgent       *config.Agent
+	InRepoConfigCache *config.InRepoConfigCache
+}
+
+type ProwYAMLGetter interface {
+	GetProwYAML(refs *prowapi.Refs) (*config.ProwYAML, error)
+}
+
+// serveGetInrepoconfig returns a ProwYAML object marshaled into JSON.
+func (mr *Moonraker) ServeGetInrepoconfig(w http.ResponseWriter, r *http.Request) {
+	body, err := io.ReadAll(r.Body)
+	if err != nil {
+		logrus.WithError(err).Info("unable to read request")
+		http.Error(w, fmt.Sprintf("bad request %v", err), http.StatusBadRequest)
+		return
+	}
+
+	refs := &prowapi.Refs{}
+	err = json.Unmarshal(body, refs)
```

Comment:
For future compatibility it might be a good idea to define a new message payload type rather than using `prowapi.Refs` directly. I'm specifically thinking that if we ever determine that we need more fields in the request payload, it would be helpful to start with a payload like
```go
type payload struct {
  Refs prowapi.Refs
}
```

Reply:
Fixed locally.